### ISSUE TITLE
NX upgrade from 18.0.3 to 18.0.4 breaks all microfrontend static remotes

### ISSUE THREAD

**Issue by ekiliqua** [nrwl/nx#27941]
### Current Behavior

I tried to update to the latest version of NX (19), but it caused several errors. One of those errors is the one I managed to locate by upgrading version by version, specifically from version 18.0.3 to version 18.0.4.

I have several applications (for example, app1, app2, app3). When I run the shell application without any application in devRemotes, everything seems to compile correctly, but when I try to access any of these applications from the shell, it always redirects to the same application (app3) without any console error or something.

I’m not sure if this is related, though I don't think so... One of the applications exposes the following modules:
![image](https://github.com/user-attachments/assets/a77214fe-9bfa-4cb5-a274-1c83507ba744)

And when I try to access one of them, I get the following error in the console:

![image](https://github.com/user-attachments/assets/3a250d26-faba-4a8f-9b99-f1779fc2eb09)

### Expected Behavior

Navigation between the applications should work correctly.

### Steps to Reproduce

upgrade 18.0.3 to version 18.0.4.

I have several applications (for example, app1, app2, app3). When I run the shell application without any application in devRemotes, everything seems to compile correctly, but when I try to access any of these applications from the shell, it always redirects to the same application (app3).

### Nx Report

```shell
Node   : 22.0.0
   OS     : darwin-arm64
   npm    : 10.5.1
   
   nx (global)        : 19.6.5
   nx                 : 18.0.4
   @nx/js             : 18.0.4
   @nx/jest           : 18.0.4
   @nx/linter         : 18.0.4
   @nx/eslint         : 18.0.4
   @nx/workspace      : 18.0.4
   @nx/angular        : 18.0.4
   @nx/cypress        : 18.0.4
   @nx/devkit         : 18.0.4
   @nx/eslint-plugin  : 18.0.4
   @nrwl/tao          : 18.0.4
   @nx/web            : 18.0.4
   @nx/webpack        : 18.0.4
   typescript         : 5.3.3
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

In this upgrade, from 18.0.3 to 18.0.4 only changes package versions. Migrations file is empty.

**Comment by Coly010** [comment]
@ekiliqua Your error suggests that something went wrong when your remote application was built. 

However, that error is also coming from older version of Nx. If you continue with your migration forward towards latest Nx, does the same error still get produced?

**Comment by ekiliqua** [comment]
@Coly010 yes, the problem is the same... The first attempt was to run 'nx migrate latest' but it was still breaking, so I went version by version to try to locate the error. If migrate to latest only change the literal of the error when try to load "./Agenda" from "utilidades" instead "63855" and the problem with routing is the same.

This "upgrade" only change files of nx from 18.0.3 to 18.0.4.

**Comment by Coly010** [comment]
@ekiliqua Can you continue from version 18.0.4 to latest following this order:

`npx nx migrate 19.0.8`
`npx nx migrate --run-migrations`
`npx nx mgrate 19.3.2`
`npx nx migrate --run-migrations`
`npx nx mgrate 19.5.7`
`npx nx migrate --run-migrations`
`npx nx mgrate latest`
`npx nx migrate --run-migrations`

Then, after all this run
`npx nx reset`

After the reset has completed, then try to serve your host again.

**Comment by ekiliqua** [comment]
@Coly010 The issue persists. After investigating a bit more into the generated remoteEntry.mjs files, they all have the same code. It seems that all of them are the same application exposed on all ports.

`var moduleMap = {
	"./Module": () => {
		return Promise.all(/*! __federation_expose_Module */[__webpack_require__.e("default-webpack_sharing_consume_default_angular_core_angular_core"), __webpack_require__.e("default-webpack_sharing_consume_default_angular_common_angular_common"), __webpack_require__.e("default-webpack_sharing_consume_default_primeng_api_primeng_api"), __webpack_require__.e("default-webpack_sharing_consume_default_rxjs_rxjs"), __webpack_require__.e("default-webpack_sharing_consume_default_angular_router_angular_router"), __webpack_require__.e("default-webpack_sharing_consume_default_XXXXXXX_domain_XXXXXXX_domain"), __webpack_require__.e("default-webpack_sharing_consume_default_ngx-translate_core_ngx-translate_core"), __webpack_require__.e("default-webpack_sharing_consume_default_XXXXXXX_data-access_XXXXXXX_data-access"), __webpack_require__.e("default-apps_provision-usuarios_src_app_remote-entry_entry_module_ts")]).then(() => (() => ((__webpack_require__(/*! apps/provision-usuarios/src/app/remote-entry/entry.module.ts */ 18133)))));
	}
};`

It seems that they all export the same: apps/provision-usuarios/src/app/remote-entry/entry.module.ts

**Comment by ekiliqua** [comment]
Hello @Coly010,

Hello, I think I have found the source of the error. It seems that the migration is done correctly, but what actually breaks the navigation is my configuration of the project.json files...

I have removed my configuration and left the default one that comes with a newly created project in version 19.

![image](https://github.com/user-attachments/assets/5f6af3a3-badb-46a0-920d-177f0cc4b74d)
![image](https://github.com/user-attachments/assets/9328380c-adf8-4707-8aa0-d1bd745a7849)

Removing the configuration seems to work; maybe when the static servers are generated, they are not being taken into account, and the redirection does not work.

**Comment by Coly010** [comment]
That makes sense. I'll see what I can do around this

**Comment by ekiliqua** [comment]
Hello @Coly010 ,

Today I migrated the project to the latest versions, removed all the outputPaths, and corrected the namedInputs from the bug I reported the other day, which were not being added automatically, and it seems that the application is working (although there are many logs in the console; I might open another discussion thread about this).

I have checked that if I only remove the outputPath from "build" but leave those inside each configuration profile, the application does not work. It's necessary to remove all of them... probably because it searches for those paths internally and doesn't find them. It would be good if the static server read the outputPath specified in the configuration and, if not found, used the default one in "options".

Thank you very much, regards.
